common_functions.sh

```diff
@@ -299,6 +295,24 @@ function get_binary_url() {
 	rm -f ${info_file}
 }
 
+# Get the short build version from the full version for this specific arch
+# $1 = full version
+function get_nightly_short_version() {
+	arch_build=$1
+	arch_full_version=$2
+	if [ "${arch_build}" = "nightly" ]; then
+		# Remove date and time at the end of full_version for nightly builds.
+		# Handle both the old and new date-time formats used by the Adopt build system.
+		# Older date-time format - 201809270034
+		arch_version=$(echo ${arch_full_version} | sed 's/-[0-9]\{4\}[0-9]\{2\}[0-9]\{2\}[0-9]\{4\}$//')
+		# New date-time format   - 2018-09-27-00-34
+		arch_version=$(echo ${arch_version} | sed 's/-[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}-[0-9]\{2\}-[0-9]\{2\}$//')
```

Comment:
Isn't this overriding the previous one?

Reply:
We just want one version at the end of it as it is either of the those date formats.

Reply:
OK, maybe update the comment on the 2nd line "If there's a newer version then use that"

Reply:
Hmm, thinking about this some more, what if there's only the old format and then line 309 executes, does that corrupt the older format?

Reply:
Sorry for the terse updates, was travelling at the time. It works for both formats
```
# Old format
$ arch_full_version="jdk8u-201809270034"
$ arch_version=$(echo ${arch_full_version} | sed 's/-[0-9]\{4\}[0-9]\{2\}[0-9]\{2\}[0-9]\{4\}$//')
$ echo $arch_version
jdk8u
$ arch_version=$(echo ${arch_version} | sed 's/-[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}-[0-9]\{2\}-[0-9]\{2\}$//')
$ echo $arch_version
jdk8u

# New Format
$ arch_full_version="jdk8u-2018-09-27-00-34"
$ arch_version=$(echo ${arch_full_version} | sed 's/-[0-9]\{4\}[0-9]\{2\}[0-9]\{2\}[0-9]\{4\}$//')
$ echo $arch_version
jdk8u-2018-09-27-00-34
$ arch_version=$(echo ${arch_version} | sed 's/-[0-9]\{4\}-[0-9]\{2\}-[0-9]\{2\}-[0-9]\{2\}-[0-9]\{2\}$//')
$ echo $arch_version
jdk8u
```
Also this code was already present all this while, I just moved it to a function as I need to call it from two places now.